dist/init/linux-systemd/caddy.service

```diff
@@ -5,18 +5,32 @@ After=network-online.target
 Wants=network-online.target systemd-networkd-wait-online.service
 
 [Service]
+; Only allow 1 restart with 10 seconds so that caddy doesn't get locked out by PAM
 Restart=on-failure
-StartLimitInterval=86400
-StartLimitBurst=5
+StartLimitInterval=10
+StartLimitBurst=1
```

Comment:
When to restart, and in which phase, that topic can fill an entire meetup. And there's *no-fault restarts*: Linux kills your process because the system is about to run out of memory, or the hardware is faulty.

In my experience most operators don't care about configuring `After=` and `Before=` correctly (`sv` with **runit**, or `/etc/rc.conf` with (I guess) **OpenRC**) and rely on restarts for missing IP addresses, mount points/directories, hardware etc. A bad practice.

With **systemd** you can tell startup phase (»starting«) from a runtime phase (»serving«) and therefore kill a process if it didn't start within some seconds, or hangs. That's been the reason for this branch last year:
https://github.com/wmark/caddy/tree/notify-init